packages/synapse-react-client/src/components/DataGrid/components/GridSessionSelectionField.tsx

```diff
@@ -0,0 +1,133 @@
+import { displayToast } from '@/components/ToastMessage/ToastMessage'
+import {
+  useDeleteGridSession,
+  useGetGridSessionsInfinite,
+} from '@/synapse-queries/grid/useGridSession'
+import { Autocomplete, Grid, MenuItem, TextField } from '@mui/material'
+import { GridSession } from '@sage-bionetworks/synapse-client'
+import noop from 'lodash-es/noop'
+import { useMemo } from 'react'
+
+type GridSessionSelectionFieldProps = {
+  inputValue: string
+  onInputValueChange: (value: string) => void
+  onConfirmInput?: (value: string) => void
+  onSessionChange?: (session: GridSession | null) => void
+  onReplicaChange?: (replicaId: number | null) => void
+}
+
+/**
+ * Utility form field to pick the grid session from existing sessions. Currently, for testing/demo purposes only.
+ */
+export default function GridSessionSelectionField(
```

Comment:
Extracted the demo/testing session selector from `StartGridSession` so I could use it in another story.